### ISSUE TITLE
Clean up and improve code related to calling 'gi.actions/identity/login'

### ISSUE THREAD

**Issue by taoeffect** [okTurtles/group-income#1765]
### Problem

In a recent commits 0a1cd00f3d47b82650ce7ecf50af2c7dda12ccb6 and 8e1701e4698bf88531cc83836b751f58e3cbda4b to `e2e-protocol`, @corrideat had to modify and complicate the way that `LoginForm.vue`'s `login` method worked in order to make sure that it waited for `'gi.actions/identity/login'` to finish loading all of the contracts.

There are 4 files in the reasons behind this:

1. `frontend/main.js` — this file calls `'gi.actions/identity/login'`, but in order to prevent a blank white screen from appearing while contracts are loading, `'gi.actions/identity/login'` was modified so that it wouldn't await on a promise that waits for all contracts to finish loading (see comment above call to `sbp('chelonia/contract/sync', contractIDs)` in `actions/identity.js`)
2. `frontend/controller/actions/identity.js` — this is where `'gi.actions/identity/login'` is defined.
3. `Join.vue` — this file is what loads `LoginForm.vue`, and it needs the login to finish loading all of the contracts so that it can get the correct and complete `userState` before calling `'gi.actions/group/joinAndSwitch'`
4. `LoginForm.vue` — This is the file that @corrideat had to complicate in order to re-introduce waiting-for-contracts-to-sync behavior to `'gi.actions/identity/login'`. We essentially want to undo those chains and have it call `await sbp('gi.actions/identity/login', ..)` as it did before without all the complicated event-listener stuff.

### Solution

There are two possible solutions. Both of them will involve restoring `LoginForm.vue` to the way that it was, and making sure to `await` on the call to `'gi.actions/identity/login'`.

#### Solution 1 (Hard)

There might be some way to modify `'gi.actions/identity/login'` so that it awaits on the call to `sbp('chelonia/contract/sync', contractIDs)`, and then inside of `main.js`, some way to call `'gi.actions/identity/login'` within the `mounted` method (but not awaiting on it), so that the UI does not freeze.

However, this might be difficult to achieve without breaking tests, as the comment in `main.js` suggests:

```js
    // NOTE: important to do this before setting up Vue.js because a lot of that relies
    //       on the router stuff which has guards that expect the contracts to be loaded
    const identityContractID = await sbp('gi.db/settings/load', SETTING_CURRENT_USER)
    try {
      if (identityContractID) {
        sbp('okTurtles.events/on', CONTRACT_IS_SYNCING, initialSyncFn)
        // TODO: 'gi.actions/identity/login' will return before the contract
        // state is synced
        // This is to continue with the following block that is setting up
        // Vue.js, but as a result the login events listened for below may
        // have already fired. Instead, this block should be moved below the
        // Vue.js set up, and the router be updated to handle not having a login
        // state initially.
        await sbp('gi.actions/identity/login', { identityContractID })
      }
```

#### Solution 2 (Easy)

The second solution is simply to pass in a boolean option to `'gi.actions/identity/login'` called `dontWaitForSync`.

The call to `sbp('chelonia/contract/sync', contractIDs)` is modified like so:

```js
const p = sbp('chelonia/contract/sync', contractIDs)...
```

And then after the catch block, just before returning from the function, we await on `p` unless `dontWaitForSync` is `true`.

Then `main.js` is modified to pass in `dontWaitForSync: true`, and the changes to `LoginForm.vue` can be reverted.

**Comment by Silver-IT** [comment]
My personal opinion is to create two identity contract functions, and to join/leave group using those two functions.
```
'gi.contracts/identity/joinGroup': () => {
  // TODO: implement...
},
'gi.contracts/identity/leaveGroup': () => {
  // TODO: implement...
},
```

This way will remove `gi.actions/identity/saveOurLoginState`, `gi.actions/identity/updateLoginStateUponLogin`, and other codes related to these two functions.
And also it would provide us easy way to sync the `loginState`, and would fix active issues like Issue #1769 (and maybe more).

cc: @corrideat, @taoeffect

**Comment by taoeffect** [comment]
@Silver-IT Yes that's an interesting idea, but it's unrelated to this issue. I remember you mentioning this before... is there an open issue or an existing comment with it? If not, feel free to create a separate issue for it.
